src/apps/btc/btc_sign.c

```diff
@@ -69,18 +70,15 @@ static uint8_t _hash_prevouts[32] = {0};
 static uint8_t _hash_sequence[32] = {0};
 
 // used during STATE_OUTPUTS
-static sha256_context_t _hash_outputs_ctx;
+static void* _hash_outputs_ctx = NULL;
 // By the end of STATE_OUTPUTS, will contain the hashOutputs hash.
 // https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki step 8.
 static uint8_t _hash_outputs[32] = {0};
 
 // hashes bytes, puts result into out. bytes and out can overlap.
 static void _sha256(const uint8_t* bytes, size_t bytes_len, uint8_t* out)
 {
-    sha256_context_t ctx;
-    sha256_reset(&ctx);
-    noise_sha256_update(&ctx, bytes, bytes_len);
-    sha256_finish(&ctx, out);
+    rust_sha256(bytes, bytes_len, out);
```

Comment:
seems like a wrapper for a single function call now. did you consider removing `_sha256` func altogether and replacing calls with `rust_sha256`?

Reply:
indeed. fixed!